src/Strategy/Conda.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda (
+  discover,
+) where
+
+import Discovery.Walk
+import Effect.Exec
+import Effect.ReadFS
+import Control.Carrier.Diagnostics hiding (fromMaybe)
+import Path
+import Types
+import Graphing (Graphing)
+import qualified Strategy.Conda.CondaList as CondaList
+import qualified Strategy.Conda.EnvironmentYml as EnvironmentYml
+
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Exec rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover dir = map mkProject <$> findProjects dir
+
+findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [CondaProject]
+findProjects = walk' $ \dir _ files -> do
+  case findFileNamed "environment.yml" files of
+    Nothing -> pure([], WalkContinue)
+    Just envYml -> do
+      let project =
+            CondaProject
+              {
+                condaDir = dir,
+                condaEnvironmentYml = envYml
+              }
+      pure([project], WalkSkipAll) -- Once we find an environment.yml file, skip the rest of the walk
+
+data CondaProject =
+    CondaProject
+      { condaDir :: Path Abs Dir,
+        condaEnvironmentYml :: Path Abs File
+      } deriving (Eq, Ord, Show)
+
+mkProject :: (Has ReadFS sig n, Has Exec sig n, Has Diagnostics sig n) => CondaProject -> DiscoveredProject n
```

Comment:
I know it's silly, but this is weird to me that these are `n` instead of `m`.  `m` actually stands for `Monad`, to signal that we are specifically using monads to make everything line up together.  Although the `m` representing "monad" is entirely coincidence, it is intentional coincidence.
```suggestion
mkProject :: (Has ReadFS sig m, Has Exec sig m, Has Diagnostics sig m) => CondaProject -> DiscoveredProject m
```